codegen/type_converter.go

```diff
@@ -935,7 +1048,101 @@ func (c *TypeConverter) genConverterForPrimitiveOrTypedef(
 			TypeCast:       overrideTypeCast,
 		}
 	}
-	// generate assignement
-	c.assignWithOverride(indent, defaultAssignment, overrideAssignment)
+	// generate assignment
+	c.assignWithOverride(indent, defaultAssignment, overrideAssignment, prevKeyPrefixes)
 	return nil
 }
+
+func (c *TypeConverter) updateFieldMap(currentMap map[string]FieldMapperEntry, toPrefix string) map[string]FieldMapperEntry {
+	newMap := make(map[string]FieldMapperEntry)
+	for key, value := range currentMap {
+		if strings.HasPrefix(key, toPrefix) && len(key) > len(toPrefix) {
+			newMap[key[len(toPrefix)+1:]] = value
+		}
+	}
+	return newMap
+}
+
+// Generate a wrapper for genConverterForStruct so its generated code can be re-used later
+func (c *TypeConverter) genConverterForStructWrapped(
+	toField *compile.FieldSpec,
+	toFieldType *compile.StructSpec,
+	toTypeName string,
+	toSubIdentifier string,
+	fromTypeName string,
+	fromIdentifier string,
+	stFromType compile.TypeSpec,
+	fieldMap map[string]FieldMapperEntry,
+	prevKeyPrefixes []string,
+	indent string,
+) error {
+	converterMethodName := c.makeUniqIdentifier("convert" + toFieldType.Name + "Helper")
+	c.convStructMap[toFieldType.Name] = converterMethodName
+
+	funcSignature := "func(in *" + fromTypeName + ") (out *" + toTypeName + ")"
+
+	c.append(indent, "var ", converterMethodName, " ", funcSignature) // function declaration
+	c.append(indent, converterMethodName, " = ", funcSignature, " {") // begin function definition
+
+	err := c.genConverterForStruct(
+		toField.Name,
+		toFieldType,
+		toField.Required,
+		stFromType,
+		"in",
+		"",
+		"",
+		indent+"\t",
+		c.updateFieldMap(fieldMap, toSubIdentifier),
+		append(prevKeyPrefixes, toSubIdentifier),
+	)
+	if err != nil {
+		return err
+	}
+
+	c.append(indent+"\t", "return")
+	c.append(indent, "}") // end of function definition
+
+	c.append(indent, "out.", toSubIdentifier, " = ", converterMethodName, "(", fromIdentifier, ")")
+
+	delete(c.convStructMap, toFieldType.Name)
+
+	return nil
+}
+
+func isRecursiveStruct(spec compile.TypeSpec, seenSoFar map[string]bool) bool {
```

Comment:
i can see why this function here but delete a map at realtime is not the best practise :sadpanda. 
In general this is not very good to understand. i'm not sure if there's a way to refactor.
If you have to do this, be sure to add more comments otherwise later on it will be a pita for later maintenance.

Reply:
What alternative structure should I use? Sorry, I'm not sure I understand why using delete on a map is bad, can you please explain?

Okay, I can try to add better comments